components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+                        }
+                        //if not having any hybrid environment give 1st environment in api published list
+                        if (environment == null) {
+                            environment = environments.get(publishedEnvironments.toArray()[0]);
+                        }
+                        String gatewayEndpoint = environment.getApiGatewayEndpoint();
+                        if (gatewayEndpoint != null && gatewayEndpoint.contains(",")) {
+                            endpoint = gatewayEndpoint.split(",")[0];
+                        } else {
+                            endpoint = gatewayEndpoint;
+                        }
+                    }
+                }
+                //String apiPath = APIUtil.getAPIPath(apiIdentifier);
+                if (endpoint.endsWith(RegistryConstants.PATH_SEPARATOR)) {
+                    endpoint.substring(0, endpoint.length() - 1);
+                }
+                // We do not need the version in the base path since with the context version strategy, the version is
+                // embedded in the context
+                String basePath = endpoint + api.getContext();
+                resourceConfig.put("basePath", basePath);
+
+                JSONArray resource_configs = (JSONArray) resourceConfig.get("apis");
+
+                //Iterating each Sub resourcePath config
+                int subResCount = 0;
+                while (subResCount < resource_configs.size()) {
+                    JSONObject subResource = (JSONObject) resource_configs.get(subResCount);
+                    String uriTempVal = (String) subResource.get("path");
+                    uriTempVal = uriTempVal.startsWith("/") ? uriTempVal : ("/" + uriTempVal);
+
+                    JSONArray operations = (JSONArray) subResource.get("operations");
+                    //Iterating each operation config
+                    for (Object operation1 : operations) {
+                        JSONObject operation = (JSONObject) operation1;
+                        String httpVerb = (String) operation.get("method");
+                        /* Right Now PATCH is not supported. Need to remove this check when PATCH is supported*/
+                        if (!"PATCH".equals(httpVerb)) {
+                            URITemplate template = new URITemplate();
+                            Scope scope = APIUtil.findScopeByKey(getScopes(resourceConfigsJSON), (String) operation.get("scope"));
+
+                            String authType = (String) operation.get("auth_type");
+                            if (authType != null) {
+                                if (authType.equals("Application & Application User")) {
+                                    authType = APIConstants.AUTH_APPLICATION_OR_USER_LEVEL_TOKEN;
+                                }
+                                if (authType.equals("Application User")) {
+                                    authType = "Application_User";
```

Comment:
APIDefinitionFromSwagger12 is used with swagger v1.2(In APIM 1.8.0) and removed since it is no longer in use(In APIM 1.9.0)
Instead we have APIDefinitionFromSwagger20 Implemented from same abstract class which implements APIDefinitionFromSwagger12